crates/pulsar-core/src/pdk/module.rs

```diff
@@ -296,22 +296,29 @@ pub struct ModuleReceiver {
 
 impl ModuleReceiver {
     /// Receive an [`Event`] from the [`Bus`].
-    pub async fn recv(&mut self) -> Result<Arc<Event>, BusError> {
-        let mut lost: u64 = 0;
-        loop {
-            match broadcast::Receiver::recv(&mut self.rx).await {
-                Ok(value) => {
-                    if lost > 0 {
-                        log::warn!(
-                            target : &self.module_name,
-                            "brodcast channel lagged {lost} messages",
-                        );
-                    }
-                    return Ok(value);
+    pub fn recv(&mut self) -> impl Future<Output = Result<Arc<Event>, BusError>> + '_ {
+        recv_with_lagged_messages(&mut self.rx, &self.module_name)
+    }
+}
+
+pub async fn recv_with_lagged_messages(
```

Comment:
Since this is now public, I would add a doc comment. Eg. `Receive an event from the bus. Log a warning if we've lost messages.`.
I would rename `recv_with_lagged_messages` to `receive_event()` or `receive_from_pulsar_bus()`

Reply:
renamed to `receive_from_broadcast`